FreeAPS/Sources/Shortcuts/Bolus/BolusIntent.swift

```diff
@@ -0,0 +1,80 @@
+import AppIntents
+import Foundation
+import Intents
+import Swinject
+
+@available(iOS 16.0,*) struct BolusIntent: AppIntent {
+    // Title of the action in the Shortcuts app
+    static var title = LocalizedStringResource("Enact Bolus", table: "ShortcutsDetail")
+
+    // Description of the action in the Shortcuts app
+    static var description = IntentDescription(.init("Allow to send a bolus to the app", table: "ShortcutsDetail"))
+
+    internal var bolusRequest: BolusIntentRequest
+
+    init() {
+        bolusRequest = BolusIntentRequest()
+    }
+
+    @Parameter(
+        title: LocalizedStringResource("Amount", table: "ShortcutsDetail"),
+        description: LocalizedStringResource("Bolus amount in U", table: "ShortcutsDetail"),
+        controlStyle: .field,
+        inclusiveRange: (lowerBound: 0, upperBound: 200),
+        requestValueDialog: IntentDialog(LocalizedStringResource(
+            "What is the value of the bolus amount in insulin units ?",
+            table: "ShortcutsDetail"
+        ))
+    ) var bolusQuantity: Double?
```

Comment:
In order to allow users to input the bolus amount (as noted by @mountrcg [here](https://github.com/nightscout/Trio/pull/177#issuecomment-2116948179)), do not mark this as optional. Achieve this by removing the `?`

So, change this line from:
`) var bolusQuantity: Double?`

To:
`) var bolusQuantity: Double`

We'll then make changes down below at line 50 to remove the `guard` and matching `else {...} ` since this is no longer optional.

See my comment [here](https://github.com/nightscout/Trio/pull/177#issuecomment-2127028568) confirming the same.

Reply:
I committed this to the branch and tested on a physical device 👍